MoreLinq.Test/IgnoreErrorsTest.cs

```diff
@@ -0,0 +1,92 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+    using System;
+
+    [TestFixture]
+    public class IgnoreErrorsTest
+    {
+        [Test]
+        public void IgnoreErrorsIsLazy()
+        {
+            new BreakingSequence<int>().IgnoreErrors(BreakingFunc.Of<Exception, bool>());
+        }
+
+        [Test]
+        public void IgnoreErrors()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => throw new TestException("True"),
+                                             () => 2,
+                                             () => throw new TestException("False"),
+                                             () => 3);
+
+            var result = source.IgnoreErrors((TestException e) => bool.Parse(e.Message));
+            var expectations = Enumerable.Range(1, 2);
+
+            Assert.That(result.Take(2), Is.EqualTo(expectations));
+            Assert.Throws<TestException>(() => result.ElementAt(2));
+        }
+
+        [Test]
+        public void IgnoreErrorsNoExceptionTypeMatch()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => 2,
+                                             () => throw new TestException(),
+                                             () => 3);
+
+            var result = source.IgnoreErrors((ArgumentException e) => true);
+            var expectations = Enumerable.Range(1, 2);
+
+            Assert.That(result.Take(2), Is.EqualTo(expectations));
+            Assert.Throws<TestException>(() => result.ElementAt(2));
+        }
+
+        [Test]
+        public void IgnoreErrorsWithBaseException()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => throw new TestException(),
+                                             () => 2,
+                                             () => throw new Exception(),
+                                             () => 3,
+                                             () => throw new NullReferenceException(),
+                                             () => 4,
+                                             () => throw new ArgumentException(),
+                                             () => 5);
+
+            var result = source.IgnoreErrors((Exception e) => true);
+            var expectations = Enumerable.Range(1, 5);
+
+            Assert.That(result, Is.EqualTo(expectations));
+        }
+
+        [Test]
+        public void IgnoreErrorsWithParserFunction()
```

Comment:
Rename to `IgnoreErrorsInParsing`, `IgnoreErrorsInIntegerParsing` or `IgnoreErrorsInInt32Parsing`.